yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -492,15 +492,30 @@ private[spark] class Client(
 
         case None =>
           // No configuration, so fall back to uploading local jar files.
-          logWarning(s"Neither ${SPARK_JARS.key} nor ${SPARK_ARCHIVE.key} is set, falling back " +
+          logInfo(s"Neither ${SPARK_JARS.key} nor ${SPARK_ARCHIVE.key} is set, falling back " +
```

Comment:
I really think this should be a warning. Falling back into this mode means your Spark app's startup time is much slower. People really should be aware that they're falling into this case, and a warning helps more than an info log, even if a tiny bit.